tests/test_modules.py

```diff
@@ -189,3 +195,28 @@ def test_circuit_composition(n_qubits: int) -> None:
     )
     truth = pyq.QuantumCircuit(n_qubits, [pyq.X([0], 1), pyq.Y([0], 1), rx, pyq.RY([0], 1)])
     assert circ == truth
+
+
+def test_CSWAP_state000_controlqubit_0() -> None:
+    result: torch.Tensor = func_pyq.operation.CSWAP(state_000, (0, 1, 2), 3)
```

Comment:
@Doomsk youre not using the pyq.modules.CSWAP?

Reply:
@dominikandreasseitz I'm just following the previous tests, using `operation.<gate>` as `Tensor` to compare the initial and final states, for example, as in `test_operations.py` line 145:

```python
def test_CRY_state01_controlqubit_0() -> None:
    result: torch.Tensor = operation.CRY(pi, state_01, (1, 0), 2)
    assert torch.allclose(state_11, result)
```

Nonetheless, they should produce the same result, since both `operation` and `modules` call from the same source on `OPERATIONS_DICT` at `utils.py`.

Reply:
yes but all the other test in test_modules use the module API? would be really nice if you can change it

Reply:
@nmheim mentioned  the API was going to be deprecated, or was it another API?

Reply:
everything in core (except utils) will be deprecated, yes. but i am asking you to use the pyqtorch/modules/ (module API), and your tests use core/operation stuff which will be deprecated